### ISSUE TITLE
open() rewrite doesn't always consider other positional args

### ISSUE THREAD

**Issue by jdufresne** [asottile/pyupgrade#312]
Minimal input:

```py
with open('blah.txt', 'r', 'utf-8') as fp:
    pass
```

Actual output:

```py
with open('blah.txt', 'utf-8') as fp:
    pass
```

Expected:

```py
with open('blah.txt', encoding='utf-8') as fp:
    pass
```

The rewrite causes the following Python error upon executing the code:

```
Traceback (most recent call last):
  File "blah.py", line 1, in <module>
    with open('blah.txt', 'utf-8') as fp:
ValueError: invalid mode: 'utf-8'
```

Discovered in a real code base:

https://github.com/andialbrecht/sqlparse/blob/28a2acdd9b307c95d2111f1f831b1a5afc634691/sqlparse/cli.py#L175

**Comment by asottile** [comment]
so uh, that code is broken, it's passing `args.encoding` to the [`buffering` parameter](https://docs.python.org/3/library/functions.html#open)

but pyupgrade shouldn't break it further

**Comment by asottile** [comment]
this should be fixed in 2.4.4 -- thanks again for the issue!
